### PR TITLE
feat: Add support for Github Enterprise

### LINKED ISSUE #125: Add support for Github Enterprise
While running this action in an Enterprise Environment, it fails to create the check because it assumes that the Github API url is ` https://api.github.com/`

Example error: 

> Error: request to https://api.github.com/repos/myOrg/myRepo/check-runs failed, reason: connect ETIMEDOUT

We should add a configuration parameter that defaults to https://api.github.com and allows users to pass an alternative base url 

See : 

https://github.com/amannn/action-semantic-pull-request/blob/6282ee339b067cb8eab05026f91153f873ad37fb/action.yml#L35-L37

https://github.com/amannn/action-semantic-pull-request/blob/348e2e6922130ee27d6d6a0a3b284890776d1f80/src/index.js#L24-L26

### PR DESCRIPTION
**Pull request by yeikel**
Closes #125

### CODE REVIEW

**Comment by ghaiszaher** [comment]
@yeikel for some reason this doesn't work, I also tried it in another branch and in a private repository, I always get `Error: Not Found`. E.g. this current PR or this one: https://github.com/ScaCap/action-surefire-report/pull/127

Did you try it with an enterprise base URL?

**Comment by yeikel** [comment]
> @yeikel for some reason this doesn't work, I also tried it in another branch and in a private repository, I always get `Error: Not Found`. E.g. this current PR or this one: #127
> 
> Did you try it with an enterprise base URL?

That's strange. In my experience 404 errors in Github tend to be about lack of permissions but in this scenario it could be something else

Yes, I tested it in a private instance of GHE and it is working as expected

**Comment by yeikel** [comment]
Can we try adding debug logs to see what is the url is trying in your test? 

See : https://actions-cool.github.io/octokit-rest/guide/12_debug/

**Comment by yeikel** [comment]
I tested the latest conditional logic pushed with 56ed3ba196403f5cd0ead2495fdf17b28719c4c1 both with Github.com: https://github.com/yeikel/action-surefire-report/actions/runs/4153632808/jobs/7185381214 and with a private instance of GHE and that seems to work

I am not entirely sure why setting the github.com API URL directly results in authentication errors though.

File: README.md [hunk 1]
```diff
@@ -45,6 +45,11 @@ Optional. Usually in form of `github_token: ${{ secrets.GITHUB_TOKEN }}`. The de
 
 Optional. Set to `true` to get the file name from the stack trace. The default is `false`.
 
+### `github_base_url`
+
+Optional : If you use GitHub Enterprise, you can set this to the URL of your server(e.g. https://github.myorg.com/api/v3). The default is https://api.github.com/
```
Comment (ghaiszaher) [inline README.md:50]:
```suggestion
Optional: If you use GitHub Enterprise, you can set this to the URL of your server (e.g. https://github.myorg.com/api/v3).
```

File: action.yml [hunk 2]
```diff
@@ -39,6 +39,11 @@ inputs:
     description: 'get filename from stack trace'
     required: false
     default: 'false'
+  github_base_url:
+    description: 'If you use Github Enterprise, you can set this to the URL of your server (e.g. https://github.myorg.com/api/v3)'
```
Comment (ghaiszaher) [inline action.yml:43]: minor
```suggestion
    description: 'if you use Github Enterprise, you can set this to the URL of your server (e.g. https://github.myorg.com/api/v3)'
```

**Review by ghaiszaher — APPROVED** [review]
Thanks for fixing!

**Comment by yeikel** [comment]
> Thanks for fixing!

Thank you for reviewing. Please let me know if you have any additional feedback 🚀

**Review by ghaiszaher — APPROVED** [review]

**Review by crypticmind — APPROVED** [review]

**Review by renatoabreu11 — APPROVED** [review]

**Pull request merged**